tuxemon/core/event/eventengine.py

```diff
@@ -449,3 +457,47 @@ def process_event(self, event):
                 self.process_map_event(map_event)
 
         return event
+
+
+@contextmanager
+def add_error_context(event, item, game):
+    """
+    :type event: core.event.EventObject
+    :type item: core.event.MapCondition or core.event.MapAction
+    :type game: core.control.Control
+    :rtype None
+    """
+    try:
+        yield
+    except Exception:
+        file_name = game.get_map_filepath()
+        tree = etree.parse(file_name)
+        event_node = tree.find("//object[@id='%s']" % event.id)
```

Comment:
When i opened this ticket i was thinking that we would store the text of each map event in the map event object, along with the map name and line number (if possible).  I think this approach of reading the source tmx file after the exception is raised is good enough for now, but in the future, i would also like this info on each event.  there is already a debug mode which lists events, and it might be useful to have the event info before crashing.